Command/ImportCommand.php

```diff
@@ -0,0 +1,174 @@
+            ->addArgument(
+                'contentTypeName',
+                InputArgument::REQUIRED,
+                'Content type name to import into'
+            )
+            ->addArgument(
+                'archive',
+                InputArgument::REQUIRED,
+                'Zip file containing the json files'
+            )
+            ->addOption(
+                'bulkSize',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Size of the elasticsearch bulk request',
+                500
+            )
+            ->addOption(
+                'raw',
+                null,
+                InputOption::VALUE_NONE,
+                'The content will be imported as is. Without any field validation, data stripping or field protection'
+            )
+            ->addOption(
+                'dont-sign-data',
+                null,
+                InputOption::VALUE_NONE,
+                'The content will not be signed during the import process'
+            )
+            ->addOption(
+                'force',
+                null,
+                InputOption::VALUE_NONE,
+                'Also treat document in draft mode'
+            )
+            ->addOption(
+                'dont-finalize',
+                null,
+                InputOption::VALUE_NONE,
+                'Don\'t finalize document'
+            )
+            ->addOption(
+                'businessKey',
+                null,
+                InputOption::VALUE_NONE,
+                'Try to identify documents by their business keys'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $contentTypeName = $input->getArgument('contentTypeName');
+        $archiveFilename = $input->getArgument('archive');
+        $bulkSize = $input->getOption('bulkSize');
+        $rawImport = $input->getOption('raw');
+        $signData = !$input->getOption('dont-sign-data');
+        $finalize = !$input->getOption('dont-finalize');
+        $force = $input->getOption('force');
+        $replaceBusinessKey = $input->getOption('businessKey');
+
+
+        $contentType = $this->contentTypeService->getByName($contentTypeName);
+        if (!$contentType instanceof ContentType) {
+            $output->writeln(sprintf('<error>Content type %s not found</error>', $contentTypeName));
+            return -1;
+        }
+
+        if ($contentType->getDirty()) {
+            $output->writeln(sprintf('<error>Content type %s is dirty. Please clean it first.</error>', $contentTypeName));
+            return -1;
+        }
+
+        if (!file_exists($archiveFilename)) {
+            $output->writeln(sprintf('<error>Archive file %s does not exist.</error>', $archiveFilename));
+            return -1;
+        }
+        
+        $output->writeln(sprintf('Start importing %s from %s', $contentType->getPluralName(), $archiveFilename));
+
+        $zip = new ZipArchive();
```

Comment:
If zip is chosen as input, use a new service for the underlying logic. The command should be thin, just as controllers.

Reply:
Not reviewing the zip extraction logic until it is extracted to a dedicated service.

Reply:
service is maybe a bit overkill: What about an hyper class: a863cd63a65a6999b149d8a774e00dc9dc99a9b5